pkgs/auto-fix-vscode-server.nix

```
@@ -1,90 +1,137 @@
+        which should allow for easy usage of extensions without Nix-specific modifications.
       '';
     };
-  };
+  });
 
-  nodejsWrapped = if enableFHS then nodejsFHS else nodejs;
+  patchELFScript = writeShellScript "patchelf-vscode-server.sh" ''
+    set -euo pipefail
+    PATH=${makeBinPath [ coreutils findutils patchelf ]}
+    INTERP=$(< ${stdenv.cc}/nix-support/dynamic-linker)
+    RPATH=${makeLibraryPath runtimeDependencies}
+    bin_dir=$1
 
-in writeShellScript "auto-fix-vscode-server.sh" ''
-  set -euo pipefail
-  PATH=${lib.makeBinPath [ coreutils findutils inotify-tools patchelf ]}
-
-  bins_dir=${installPath}/bin
-  node_interp=$(patchelf --print-interpreter ${nodejs}/bin/node)
-  node_rpath=$(patchelf --print-rpath ${nodejs}/bin/node)
+    # NOTE: We don't log here because it won't show up in the output of the user service.
 
-  patch_bin() {
-    local bin_dir=$1 interp
-    ln -sfT ${nodejsWrapped}/bin/node "$bin_dir/node"
-    while read -rd ''' bin; do
+    while read -rd ''' elf; do
       # Check if binary is patchable, e.g. not a statically-linked or non-ELF binary.
-      if ! interp=$(patchelf --print-interpreter "$bin" 2>/dev/null); then
+      if ! interp=$(patchelf --print-interpreter "$elf" 2>/dev/null); then
         continue
       fi
+
       # Check if it is not already patched for Nix.
-      if [[ $interp == "$node_interp" ]]; then
+      if [[ $interp == "$INTERP" ]]; then
         continue
       fi
-      patchelf --set-interpreter "$node_interp" --set-rpath "$node_rpath" "$bin"
-      patchelf --shrink-rpath "$bin"
+
+      # Patch the binary based on the binary of Node.js,
+      # which should include all dependencies they might need.
+      patchelf --set-interpreter "$INTERP" --set-rpath "$RPATH" "$elf"
+
+      # The actual dependencies are probably less than that of Node.js,
+      # so shrink the RPATH to only keep those that are actually needed.
+      patchelf --shrink-rpath "$elf"
     done < <(find "$bin_dir" -type f -perm -100 -printf '%p\0')
+
+    # Mark the bin directory as being fully patched.
+    echo 1 > "$bin_dir/.patched"
+  '';
+
+in writeShellScript "auto-fix-vscode-server.sh" ''
+  set -euo pipefail
+  PATH=${lib.makeBinPath [ coreutils findutils inotify-tools ]}
+  bins_dir=${installPath}/bin
+
+  patch_bin_dir () {
+    local bin_dir=$1
+
+    if [[ -e $bin_dir/.patched ]]; then
+      return 0
+    fi
+
+    echo "Patching Node.js of VS Code server installation in $bin_dir..." >&2
+
+    ${optionalString (nodejs != null) ''
+      ln -sfT ${if enableFHS then nodejsFHS else nodejs}/bin/node "$bin_dir/node"
+    ''}
+
+    ${optionalString (!enableFHS) ''
+      mv "$bin_dir/node" "$bin_dir/node.orig"
+      cat <<EOF > "$bin_dir/node"
+      #!/usr/bin/env bash
+
+      # Patch the VS Code server installation only if it is not already patched.
+      if ! (( \$(< '$bin_dir/.patched') )); then
```

By the time `$bin_dir` is expanded, it's already a broken path. Here's why:

When VSCode server is first being installed into `~/.vscode-server/bin/`, its directory initially is called something like

```
441438abd1ac652551dbe4d408dfcec8a499b8bf-1676342464
```
before being renamed
```
441438abd1ac652551dbe4d408dfcec8a499b8bf
```
The problem is that the auto-fix service detects the former, and so the wrapped node script won't run.

```suggestion
      if ! (( \$(< './.patched') )); then
```

---

Wait, relative paths won't work because the working directory is not guaranteed to be the bin directory.
What if we modify the `inotifywait` part of the service to hold off on patching anything until the server directory is renamed?

---

I assume has to do with your WSL setup that this happens differently, as otherwise it should be broken with my setup to. I am working from Win10 to NixOS (different machine). I could indeed probably complicate stuff with checking for that rename event, but considering this isn't happening in all setup, I suggest we just try do some string manipulation to get the first hash only.